ragna/core/_rag.py

```diff
@@ -188,7 +196,17 @@ async def prepare(self) -> Message:
                 detail=RagnaException.EVENT,
             )
 
-        await self._run(self.source_storage.store, self.documents)
+        from ragna.core import Document
+        if type(self.source_storage).__ragna_input_type__ == Document:
```

Comment:
Especially with the upcoming change for the chunking as well, I would prefer something like

```python
input = self.documents
if not issubclass(self.source_storage.__ragna_input_type__, Document):
    input = self.chunking_model(input)
if not issubclass(self.source_storage.__ragna_input_type__, Document):
    input = self.embedding_model(input)
await self._run(self.source_storage.store, input)
```